src/daft-io/src/google_cloud.rs

```diff
@@ -409,7 +409,7 @@ impl ObjectSource for GCSSource {
         page_size: Option<i32>,
         limit: Option<usize>,
         io_stats: Option<IOStatsRef>,
-    ) -> super::Result<BoxStream<super::Result<FileMetadata>>> {
+    ) -> super::Result<BoxStream<'static, super::Result<FileMetadata>>> {
```

Comment:
might be worth to make `BoxStream<'static, super::Result<FileMetadata>>` a typedef